pennylane/transforms/hamiltonian_expand.py

```diff
@@ -197,3 +201,205 @@ def processing_fn(res):
         return qml.math.convert_like(summed_dot_products, res[0])
 
     return tapes, processing_fn
+
+
+# pylint: disable=too-many-branches, too-many-statements
+def sum_expand(qscript: QuantumScript, group=True):
+    """Splits a quantum script measuring a Sum expectation into multiple scripts of summand
+    expectations, and provides a function to recombine the results.
+
+    Args:
+        qscript (.QuantumScript): the quantum script used when calculating the expectation value
+            of the Hamiltonian
+        group (bool): Whether to compute disjoint groups of Pauli observables acting on different
+            wires, leading to fewer tapes.
+
+    Returns:
+        tuple[list[.QuantumScript], function]: Returns a tuple containing a list of
+        quantum scripts to be evaluated, and a function to be applied to these
+        script executions to compute the expectation value.
+
+    **Example**
+
+    Given a Sum operator,
+
+    .. code-block:: python3
+
+        S = qml.op_sum(qml.prod(qml.PauliY(2), qml.PauliZ(1)), qml.s_prod(0.5, qml.PauliZ(2)), qml.PauliZ(1))
+
+    and a tape of the form,
+
+    .. code-block:: python3
+
+        with qml.tape.QuantumTape() as tape:
+            qml.Hadamard(wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PauliX(wires=2)
+
+            qml.expval(S)
+            qml.expval(qml.PauliZ(0))
+            qml.expval(qml.PauliX(1))
+            qml.expval(qml.PauliZ(2))
+
+    We can use the ``sum_expand`` transform to generate new tapes and a classical
+    post-processing function to speed-up the computation of the expectation value of the `Sum`.
+
+    >>> tapes, fn = qml.transforms.sum_expand(tape, group=False)
+    >>> for tape in tapes:
+    ...     print(tape.measurements)
+    [expval(PauliZ(wires=[0]))]
+    [expval(PauliY(wires=[2]) @ PauliZ(wires=[1]))]
+    [expval(PauliZ(wires=[2]))]
+    [expval(PauliZ(wires=[1]))]
+    [expval(PauliX(wires=[1]))]
+
+    Five tapes are generated: the first three contain the summands of the `Sum` operator,
+    and the last two contain the remaining observables. Note that the scalars of the scalar products
+    have been removed. These values will be multiplied by the result obtained from the execution.
+
+    We can evaluate these tapes on a device:
+
+    >>> dev = qml.device("default.qubit", wires=3)
+    >>> res = dev.batch_execute(tapes)
+
+    Applying the processing function results in the expectation value of the Hamiltonian:
+
+        >>> fn(res)
+    [-0.5, 0.0, 0.0, -0.9999999999999996]
+
+    Fewer tapes can be constructed by grouping commuting observables. This can be achieved
+    by the ``group`` keyword argument:
```

Comment:
```suggestion
    Fewer tapes can be constructed by grouping observables acting on different wires. This can be achieved
    by the ``group`` keyword argument:
```